lib/stripe/subscriptions/coupon.ex

```diff
@@ -124,12 +118,12 @@ defmodule Stripe.Coupon do
   @doc """
   List all coupons.
   """
-  @spec list(params, Stripe.options) :: {:ok, Stripe.List.of(t)} | {:error, Stripe.Error.t}
+  @spec list(params, Stripe.options()) :: {:ok, Stripe.List.of(t)} | {:error, Stripe.Error.t()}
         when params: %{
-               created: Stripe.date_query,
-               ending_before: t | Stripe.id,
-               limit: 1..100,
-               starting_after: t | Stripe.id
+               created: Stripe.date_query() | nil,
+               ending_before: t | Stripe.id() | nil,
+               limit: 1..100 | nil,
+               starting_after: t | Stripe.id() | nil
```

Comment:
I don't think the `| nil` is the right approach here. Look to the instances where we use `optional`, for example in `Stripe.Charge`.